web/styles/zulip.css

```diff
@@ -649,13 +649,20 @@ li,
     &:hover svg path {
         fill: var(--color-fill-hover-copy-icon);
     }
+    backdrop-filter: blur(4px);
+
+    mask-image: radial-gradient(
+        farthest-corner,
+        hsl(0deg 0% 0% / 100%),
+        hsl(0deg 0% 0% / 90%)
+    );
 }
 
 .copy_message {
     float: right;
     position: relative;
     cursor: pointer;
-    top: 30px;
+    top: 31px;
```

Comment:
This change is done so that the button doesn't hide the border of message with it's blur
| Before | After |
| ---- | ------ |
| ![image](https://github.com/zulip/zulip/assets/91622060/1983466f-9b3b-459c-9864-28aa1330c2cc) | ![image](https://github.com/zulip/zulip/assets/91622060/703baaa5-3903-4332-932f-4c8992c2d2d8) |

Reply:
This needs some additional fiddling. In testing, the `background-clip: content-box` property means that only the area immediately under the icon will be blurred. There should likely be a more generous blurred area beyond the edge of the icon (which will mean adjusting width or padding so that the icon sits more centered in the middle of the box).

Reply:
I've changed the padding and this is what it looks like now.
![image](https://github.com/zulip/zulip/assets/91622060/411dadc7-fb3d-44a0-a162-62100aebda58)

Reply:
This still doesn't match [the example I had in mind](https://github.com/zulip/zulip/pull/29247/files#diff-09554f973098c75bc832a46da33f2c281f593c69feaf28d196a18db6adc1190bR730), which uses an `::after` pseudo-element to make sure that the radial-gradient softens the background. Yours still has a hard edge.

Reply:
@karlstolley sorry for the late reply, this comment somehow got missed by me.
I am not using `::after` here because in case of `code-buttons-container` it has a button inside it after which the pseudo-element is generated but in this case the `div` itself is the button.
If we want to use the `::after` I think the following changes need to be made, are these changes correct?

```hbs
<div class="copy_message tippy-zulip-tooltip" data-tippy-content="{{t 'Copy and close' }}" aria-label="{{t 'Copy and close' }}">
    <button class="copy_button_base">
        {{> copy_to_clipboard_svg }}
    </button>
</div>
```
```css
.copy_message {
    float: right;
    position: relative;
    cursor: pointer;
    top: 31px;
    /* To make sure the svg doesn't occupy any vertical space. */
    margin-top: -30px;
    right: 2px;
    padding-top: 2px;

    #clipboard_image {
        margin: 0;
    }

    ::after {
        content: " ";
        backdrop-filter: blur(4px);
        position: absolute;
        height: 100%;
        width: 100%;
        z-index: -1;
        mask-image: radial-gradient(
            farthest-corner,
            hsl(0deg 0% 0% / 100%),
            hsl(0deg 0% 0% / 90%)
        );
    }
    z-index: 2;
}
```

Reply:
I can't tell by looking at them. Have you tested in a commit on your PR? Put that up and let me look...the main thing is to replicate the pattern in #29247--it might even be worth extracting or refactoring some classes there so we don't need to duplicate these delicate styles all over again.